app/src/main/java/org/mozilla/tv/firefox/components/ApplicationLifecycleSessionCache.kt

```diff
@@ -0,0 +1,63 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.tv.firefox.components
+
+import android.support.annotation.VisibleForTesting
+import android.support.annotation.VisibleForTesting.PRIVATE
+import mozilla.components.browser.session.Session
+import mozilla.components.browser.session.SessionManager
+
+/**
+ * A object that caches [Session]s in place of the [SessionManager] after the
+ * [mozilla.components.concept.engine.EngineView] owner's lifecycle has ended and restores the sessions when the
+ * EngineView is reattached. This is necessary because Android Components has a memory leak where [Session]s will keep a
+ * reference to their system WebView. They null the WebView reference when [Session]s are removed from the
+ * [SessionManager] so, as a workaround, we manually remove the [Session]s at the end of the lifecycle and re-add them
+ * when a new lifecycle begins.
+ *
+ * See more details in the issue mentioned below:
+ * TODO: implement the components solution in android-components#1915
+ *
+ * We don't use lifecycle observers in this class because each method must be called at a precise time (see each
+ * lifecycle method's kdoc for details).
+ *
+ * TODO
+ * - test clear data, on device.
+ * - Check session repo for correctness.
+ * - Will all our onDestroy methods work if the WebView is hidden in onStop?
```

Comment:
Is this TODO still valid?

Reply:
No: I forgot to commit these deletions apparently.